File: src/main/kotlin/com/kingdom/model/cards/DeckConverter.java
package com.kingdom.model.cards;

import javax.persistence.AttributeConverter;
import javax.persistence.Converter;

@Converter(autoApply = true)
public class DeckConverter implements AttributeConverter<Deck, String> {
    @Override
    public String convertToDatabaseColumn(Deck deck) {
        return deck.toString();
    }

    @Override
    public Deck convertToEntityAttribute(String s) {
        return Deck.valueOf(s);
    }
}

File: src/main/kotlin/com/kingdom/model/cards/DeckConverter.kt
package com.kingdom.model.cards

import javax.persistence.AttributeConverter
import javax.persistence.Converter

@Converter(autoApply = true)
class DeckConverter : AttributeConverter<Deck, String> {

    override fun convertToDatabaseColumn(deck: Deck): String {
        return deck.toString()
    }

    override fun convertToEntityAttribute(s: String): Deck {
        return Deck.valueOf(s)
    }
}
